### PR TITLE
[MNG-7611] Change semantics of plugin descriptor's "requiredJavaVersion"

### PR DESCRIPTION
**Pull request by kwin**
Always assume version constraint (the same as in requiredMavenVersion)

### CODE REVIEW

File: maven-core/src/main/java/org/apache/maven/plugin/internal/MavenPluginJavaPrerequisiteChecker.java [hunk 1]
```diff
@@ -18,26 +18,57 @@
...
+            constraint = versionScheme.parseVersionConstraint(requiredVersion);
+        } catch (InvalidVersionSpecificationException e) {
+            throw new IllegalArgumentException(
+                    "Invalid requiredJavaVersion given in plugin descriptor: " + e.getMessage(), e);
+        }
+        Version current;
+        try {
+            current = versionScheme.parseVersion(currentVersion);
+        } catch (InvalidVersionSpecificationException e) {
+            throw new IllegalStateException("Could not parse current Java version: " + e.getMessage(), e);
```
Comment (michael-o) [inline maven-core/src/main/java/org/apache/maven/plugin/internal/MavenPluginJavaPrerequisiteChecker.java:67]: Please don't provide duplicate messages with `e.getMessage(), e`.

**Review by michael-o — CHANGES_REQUESTED** [review]

File: maven-core/src/main/java/org/apache/maven/plugin/internal/MavenPluginJavaPrerequisiteChecker.java (hunk 1 above)
Comment (kwin) [reply maven-core/src/main/java/org/apache/maven/plugin/internal/MavenPluginJavaPrerequisiteChecker.java]: fixed in https://github.com/apache/maven/pull/888/commits/c407bbb997ed472a414d24e5d4c2b065984c6f31.

Comment (gnodet) [reply maven-core/src/main/java/org/apache/maven/plugin/internal/MavenPluginJavaPrerequisiteChecker.java]: Why not ? For the record, I would argue that gives a hint to the cause without having to read the full stack trace. Maybe even `("xxx" + e, e)` can be useful some time...

**Review by gnodet — APPROVED** [review]

File: maven-core/src/main/java/org/apache/maven/plugin/internal/MavenPluginJavaPrerequisiteChecker.java (hunk 1 above)
Comment (michael-o) [reply maven-core/src/main/java/org/apache/maven/plugin/internal/MavenPluginJavaPrerequisiteChecker.java]: When you look at our codebase and stack traces you will see duplicate messages all over. It is a pain to filter out the relevant information.

**Comment by jira-importer** [comment]
Resolve #8841

**Pull request merged**